### PR TITLE
Add on-a-stick mode

### PR DESCRIPTION
**Pull request by dpino**
The on-a-stick mode allows to run the lwAFTR in one single NIC. Incoming traffic is categorized per packet type (v4 or v6) and forwarded to a the corresponding link. The lwAFTR runs normally on each incoming link. Outgoing packets are joined together and redirected to the NIC so they can be transmitted.

The monitor program is used to set the IPv4 matching address when a lwAFTR process is running in mirroring mode. Example:

<pre>
$ sudo ./snabb lwaftr run -v --conf lwaftr.conf --on-a-stick 83:00.0 --mirror tap0
</pre>

<pre>
$ sudo ./snabb lwaftr monitor 193.5.1.253
Mirror address set to '193.5.1.253'
</pre>

Matching packets will be copied to the tap interface set by the mirroring mode.

Tap0:

<pre>
$ sudo tcpdump -qns 0 -e -i tap0
02:aa:aa:aa:aa:aa > 02:99:99:99:99:99, IPv4, length 510: 193.5.1.253.59392 > 10.10.10.1.12345: UDP, length 468
02:aa:aa:aa:aa:aa > 02:99:99:99:99:99, IPv4, length 510: 193.5.1.253.60416 > 10.10.10.1.12345: UDP, length 468
02:aa:aa:aa:aa:aa > 02:99:99:99:99:99, IPv4, length 510: 193.5.1.253.61440 > 10.10.10.1.12345: UDP, length 468
02:aa:aa:aa:aa:aa > 02:99:99:99:99:99, IPv4, length 510: 193.5.1.253.62464 > 10.10.10.1.12345: UDP, length 468
</pre>

### CODE REVIEW

**Comment by dpino** [comment]
Moved to https://github.com/Igalia/snabb/pull/381

File: src/apps/lwaftr/v4v6.lua [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+local function mirror_ipv6 (pkt, output, ipv4_num)
+   local ipv6_hdr = get_ethernet_payload(pkt)
+   local ipv4_hdr = get_ipv6_payload(ipv6_hdr)
+   if get_ipv4_dst_num(ipv4_hdr) == ipv4_num or
+         get_ipv4_src_num(ipv4_hdr) == ipv4_num then
+      transmit(output, packet.clone(pkt))
+   end
+end
+
+v4v6 = {}
```
Comment (kbara) [inline src/apps/lwaftr/v4v6.lua]: We usually seem to use an initial-caps convention.

File: src/program/lwaftr/monitor/README [hunk 2]
```diff
@@ -0,0 +1,18 @@
+Usage:
+    monitor [IPV4_ADDRESS] [PID]
+
+    -h, --help
+                                Print usage information.
+
+Optional arguments:
+
+    IPV4_ADDRESS                IPv4 address. Default: 0.0.0.0.
```
Comment (kbara) [inline src/program/lwaftr/monitor/README]: I'd change the text to "IPv4 address to mirror".

**Comment by dpino** [comment]
Updated PR:
- Renamed 'v4v6' app to 'V4V6'
- Fixed 'lwaftr monitor' README.

**Comment by kbara** [comment]
LGTM

**Pull request closed without merging**